module/state_synchronization/indexer/collection_executed_metric.go

```diff
@@ -73,6 +76,11 @@ func (c *CollectionExecutedMetricImpl) BlockFinalized(block *flow.Block) {
 	now := time.Now().UTC()
 	blockID := block.ID()
 
+	// Initialize the transactions slice for this block ID if it doesn't already exist
+	if _, exists := c.blockTransactions[blockID]; !exists {
+		c.blockTransactions[blockID] = []flow.Identifier{}
+	}
```

Comment:
nit: you shouldn't have to do this. a nil slice is treated the same as an empty slice